src/pages/costModels/costModel/updateMarkupDialog.tsx

```diff
@@ -32,23 +38,46 @@ interface Props extends WithTranslation {
 
 interface State {
   markup: string;
+  origSign: string;
+  sign: string;
 }
 
 class UpdateMarkupModelBase extends React.Component<Props, State> {
   constructor(props) {
     super(props);
+    const initialMarkup = this.props.current.markup.value;
+    const initialSign = initialMarkup.startsWith('-') ? '-' : '+';
+    const noSignValue = initialMarkup.startsWith('-') ? initialMarkup.substring(1) : initialMarkup;
```

Comment:
It appears we're parsing for "+" or "-" strings in multiple places. Instead of parsing strings, it would be cleaner to test if the markup is negative or positive. 

Perhaps this could be defined once, for example: `hasDiscount = Number(markup) < 0;`

Please see my radio button comments below

Reply:
IN some cases we are parsing what state.sign is set to and then based on that we are either adding a sign to the markup or not.  Tese test are needed now as we no longer except the sign to be adding into the text area.

Reply:
Since we're using the radio buttons, the text input shouldn't accept a negative value

<img width="542" alt="Screen Shot 2021-07-20 at 12 17 17 PM" src="https://user-images.githubusercontent.com/17481322/126359502-ee0753e9-77a5-4c5d-a43a-fd54c72b0eb7.png">

Reply:
For example, the text input can ignore certain characters.

https://stackoverflow.com/questions/14806200/disable-some-characters-from-input-field

Reply:
For example, the text input can ignore certain characters.

```
onKeyUp={this.lettersOnly}
...
private lettersOnly = event => {
    const regex = /^[0-9]*$/;
    event.target.value = event.target.value.replace(regex, '');
  }
```

Reply:
Done

Reply:
Nice trick